packages/plans-grid/src/plans-accordion/index.tsx

```diff
@@ -127,7 +133,8 @@ const PlansAccordion: React.FunctionComponent< Props > = ( {
 								isPrimary
 								isSelected={
 									!! selectedPlanProductId &&
-									primaryPlan.productIds.indexOf( selectedPlanProductId ) > -1
+									selectedPlanProductId ===
+										getPlanProduct( primaryPlan.periodAgnosticSlug, billingPeriod )?.productId
```

Comment:
comparing with the correct `productId` (and making sure they're not both `undefined`)